### ISSUE TITLE
Error while running `GeoLiftPowerFinder`

### ISSUE THREAD

**Issue by LeoUbbiali** [facebookincubator/GeoLift#14]
Hey guys,

I've been trying out the package today and I am encountering an error when  running the `GeoLiftPowerFinder` function 

## Context
-  weekly data 
- `location_id` are postcodes
- result_search screenshot

![Screenshot 2021-11-04 at 12 00 04](https://user-images.githubusercontent.com/65649681/140310380-9a9b25e6-0105-45cb-babc-75033482f2cf.jpg)

## Code

```
resultsNum <- NumberLocations(data = GeoTestData_PreTest,
                              Y_id = "Y",
                              location_id = "location",
                              time_id = "time",
                              n_sim = 500,
                              treatment_periods = 6, #6 weeks
                              plot = TRUE,
                              power = 0.8,
                              alpha = 0.1,
                              fixed_effects = TRUE,
                              ProgressBar = TRUE)

#Zipcode test Selection
resultsSearch <- GeoLiftPower.search(data = GeoTestData_PreTest,
                                     treatment_periods = c(6),
                                     N = c(2),
                                     horizon = -50, #six weeks
                                     Y_id = "Y",
                                     location_id = "location",
                                     time_id = "time",
                                     top_results = 2,
                                     alpha = 0.1,
                                     type = "pValue",
                                     fixed_effects = TRUE,
                                     ProgressBar = TRUE)

head(resultsSearch,5)

#MDE
resultsFind <- GeoLiftPowerFinder(data = GeoTestData_PreTest,
                                  treatment_periods = c(6),
                                  N = c(2),
                                  Y_id = "Y",
                                  location_id = "location",
                                  time_id = "time",
                                  effect_size = seq(0.1),
                                  top_results = 1,
                                  alpha = 0.1,
                                  model = "Ridge",
                                  fixed_effects = TRUE,
                                  ProgressBar = TRUE,
                                  plot_best = TRUE)
```

## Error encountered 

changing the `effect_size` parameter

![image](https://user-images.githubusercontent.com/65649681/140310915-d31ee326-9043-45ad-b312-1ca77395cbfa.png)

![image](https://user-images.githubusercontent.com/65649681/140310952-83004131-966f-4800-8d95-5415784f5148.png)

**Comment by NicolasMatrices-v2** [comment]
Hi Leo! Thanks for raising this issue.

I think that this may be because `GeoLiftPowerFinder` is not finding any setups that have MDEs in the range you provided for effect_size.  Out of the three alternatives, I would go with the first one (the one that has `effect_size = seq(0, 0.5, 0.05)` and change it for `effect_size = seq(0, 1, 0.2)`.  That way you have a wider range of potential effects to try out.  Please check it out and see if that solves this issue.

If that's not the case, you could also try increasing the amount of potential locations in the treatment (i.e. `N = c(2, 4, 6)`.

**Comment by ArturoEsquerra** [comment]
Hi @LeoUbbiali! Complementing Nico's response, I would recommend playing around the `effect_size` parameter in the second and third error message you provided as an example. This parameter is meant to provide a sequence of possible lifts or effect sizes that our model will go through and assess their power. It's always good to include zero in this sequence as this allows us to take a closer look at what happens when there is no simulated lift and discard combinations that are finding a significant effect when, in fact, there was no lift. Specifically, the second and third examples are missing the third argument of `seq(from, to, by)` and are returning a single effect size (0.5 and 1 respectively).

**Comment by LeoUbbiali** [comment]
Hey @NicolasMatrices-v2  and @ArturoEsquerra thank you so much for your help!

**update**: I could not solve the issue

**TLDR**:
- As the volume of 'Y' (conversions)  is very low It now makes sense that the MDE should be pretty high to detect an effect, ceteris paribus (see below the time series plot of my data)
-  however, even by setting  `effect_size = seq(0, 1, 0.05)` I get the same error (I guess it's more a "no valuable results" flag than an error :) ) 
![trend ](https://user-images.githubusercontent.com/65649681/140942134-52502338-6cd4-4cff-80c2-7415ec1410ec.png)


**further context**
- N geos = 15 (appreciate it's belowe the Best Practice)
- there might be missing row in the dataset (Y = 0 for a geo) --> should I backfill those rows with Y = 0? I guess it is handled by the package as `GeoLiftPower.search` works fine.


![image](https://user-images.githubusercontent.com/65649681/140943592-0e00b0e3-73d3-4407-a525-849bd65e85e7.png)

**Comment by NicolasMatrices-v2** [comment]
Leo, could you try incresing the potential effect size? Try changing it to `effect_size = seq(0, 50, 1)`.

**Comment by LeoUbbiali** [comment]
Thanks Nicolas, Already tried also with `effect_size = seq(0, 300, 1)`. same output

side note: I've added "missing row" in the panel data using `tidyverse::complete()"and re-run the pipeline --> no change.

this is the output of `GeoLiftPower.search` 

![image](https://user-images.githubusercontent.com/65649681/140950577-ae25cda3-1466-4e42-af9f-bfa20b4327ed.png)

**Comment by NicolasMatrices-v2** [comment]
Ok, could you try plotting out the power curve with GeoLiftPower for the combination of locations that is ranked first in the output of GeoLiftPower.search? Let's see what that looks like to have some idea of the power you could achieve with a combination of locations.

Also, in the plot above you only showed 4 locations, could you double check if you have 15 or 4 locations after GeoDataRead? Thanks

**Comment by ArturoEsquerra** [comment]
One quick additional recommendation is to make sure you have no missing values in the dataset (also make sure that each date has a value for each location - even if it's zero). We intentionally left the treatment of missing values and missing observations to the user since there are a lot of reasons why we could have these values. It could mean that Y=0, that the data was missing randomly, or even that there's a structural pattern/cause to those NA's. So, since each one of those have very specific consequences to the model we didn't want to assume the nature of the missing values.

**Comment by LeoUbbiali** [comment]
Ok guys I've managed to get through it.
Thank you so much for your help and patience @ArturoEsquerra @NicolasMatrices-v2 

It was definitely a missing value in the panel data set.

Suggestions (happy to contribute with on both options):

1. In the documentation add walk-through of the data set completion using this wrapper  [tidyr:complete()](https://tidyr.tidyverse.org/reference/complete.html)
2. tidyr:complete() embed the ` wrapper in the `GeoDataRead` function 

If we go for option two, I am happy to pull up a PR

Best,
Leo

**Comment by ArturoEsquerra** [comment]
Hi @LeoUbbiali!

Thank you very much for your suggestion! While we purposely left the treatment of missing values completely up to the user, I think that the issue you encountered could be pretty common, so we will definitely take a closer look into this topic. 

Thanks once again!
